web-api/src/storage/postgres.rs

```diff
@@ -676,22 +752,24 @@ impl Database {
         tx: &mut Transaction<'_, Postgres>,
         user_id: &UserId,
         time: DateTime<Utc>,
-        interactions: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = &DocumentId>>,
+        interactions: impl IntoIterator<IntoIter = impl ExactSizeIterator<Item = &SnippetId>>,
     ) -> Result<(), Error> {
         let mut interactions = interactions.into_iter().peekable();
         //FIXME micro benchmark and chunking+persist abstraction
         let persist = interactions.len() < 10;
 
-        let mut builder =
-            QueryBuilder::new("INSERT INTO interaction (doc_id, user_id, time_stamp) ");
+        let mut builder = QueryBuilder::new(
+            "INSERT INTO interaction (document_id, snippet_idx, user_id, time_stamp) ",
+        );
         while interactions.peek().is_some() {
             builder
                 .reset()
                 .push_values(
                     interactions.by_ref().take(Database::BIND_LIMIT / 3),
```

Comment:
🔧 the bounded arguments increased by one
```suggestion
                    interactions.by_ref().take(Database::BIND_LIMIT / 4),
```